ui/static/js/app.js

```diff
@@ -115,11 +128,13 @@ function markPageAsRead() {
 }
 
 // Handle entry status changes from the list view and entry view.
-function handleEntryStatus(element) {
+function handleEntryStatus(element, setToRead) {
     let toasting = !element;
     let currentEntry = findEntry(element);
     if (currentEntry) {
-        toggleEntryStatus(currentEntry, toasting);
+        if (!setToRead || currentEntry.querySelector("a[data-toggle-status]").dataset.value == "unread") {
```

Comment:
`dataset` returns a key/value pair, `dataset.value` doesn't seems to be the standard way of doing that. There are multiple data attributes on this element.

- [https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*)
- https://html.spec.whatwg.org/multipage/dom.html#domstringmap

Reply:
that part was copied from https://github.com/miniflux/v2/blob/de337c99e0fb0e7ad21e6a622fd777caf972e58e/ui/static/js/app.js#L149, so it's probably wrong there too

Reply:
Yes. That needs to be changed.

Reply:
turns out it's fine, there's a `data-value` attribute on the elements that contains `"read"` or `"unread"`.

Reply:
Right, I forgot about that :)